src/util.cpp

```diff
@@ -1612,8 +1612,9 @@ QMap<QString, QString> CLocale::GetAvailableTranslations()
         // get alias of translation file
         const QString strCurFileName = DirIter.next();
 
-        // extract only language code (must be at the end, separated with a "_")
-        const QString strLoc = strCurFileName.right ( strCurFileName.length() - strCurFileName.indexOf ( "_" ) - 1 );
+        // extract only language code "xx_XX" from "translation_xx_XX.qm"
```

Comment:
Somehow I feel this is an ideal candidate for using a regexp to extract the value.  It should be code that's only run through once on execution, so any overhead can be ignored.  It would be more readable and verifiable that way, I think, than trying to work out what's going on here.

Reply:
I think a regex is just slightly overkill for this. All we are doing is extracting the part between the first `'_'` and the first `'.'`, so two simple `indexOf()` calls and a `mid()` are fine. The change was needed because the resource file created on the fly by `qmake` includes the `.qm` string in the generated `.qrc` file, when we didn't include it manually in `resources.qrc`.